radsat-sk/operation/subsystems/RBattery.c

```diff
@@ -0,0 +1,279 @@
+	}
+	// Get ADC Temperatures
+	dataStorage->motherboardTemp = (ADC_COUNT_TO_MOTHERBOARD_TEMP_SCALE * storedTempData[0])
+			- ADC_COUNT_TO_MOTHERBOARD_TEMP_SHIFT;
+
+	dataStorage->daughterboardTemp1 = (ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SCALE * storedTempData[1])
+				- ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SHIFT;
+
+	dataStorage->daughterboardTemp2 = (ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SCALE * storedTempData[2])
+				- ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SHIFT;
+
+	dataStorage->daughterboardTemp3 = (ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SCALE * storedTempData[3])
+				- ADC_COUNT_TO_DAUGHTERBOARD_TEMP_SHIFT;
+
+	return 0;
+}
+
+
+/**
+ * Send a telecommand via I2C to check on the current battery voltage, and if under 6.5V, raise the global safeFlag
+ * @return nothing.
+ */
+int checkSafeFlag(void){
+
+	// Create temporary variables for sending and receiving I2C data
+	uint8_t i2c_command[BATTERY_TELEM_COMMAND_LENGTH] = {0};
+	uint8_t i2c_data[BATTERY_RESPONSE_LENGTH] = {0};
+
+	memcpy(i2c_command, &batteryVoltageCommandBytes[0], BATTERY_TELEM_COMMAND_LENGTH);
+
+	// Get the battery output voltage
+	int error = batteryTalk(i2c_command, i2c_data);
+
+	if(error != 0){
+		// TODO: Some Error Raising here?
+		// return the error
+		return error;
+	}
+
+	float converted_value = ADC_COUNT_TO_BATTERY_BUS_OUTPUT_VOLTAGE * (float)(*i2c_data);
+
+	// If the voltage is less than 6.5V then raise the safeFlag to send the cubeSat into safe mode.
+	// TODO: generate an error?
+	if(converted_value < 6.5){
+		safeFlag = 1;
+	}
+	else{
+		safeFlag = 0;
+	}
+
+	return 0;
+}
+
+/***************************************************************************************************
+                                         PRIVATE FUNCTIONS
+***************************************************************************************************/
+
+static int batteryTalk(uint8_t* command, uint8_t* response) {
+
+	// Create a temporary variable for receiving I2C data
+	// char* i2c_data = {0};
+
+	// Check for null pointers
+	if(command == NULL || response == NULL){
+		return E_INPUT_POINTER_NULL;
+	}
+
+	// Create a temp variable for passing the proper command length
+	uint8_t comm_length;
+	uint8_t comm_delay;
+
+	// Variable selection for command length dependent on if it is a telemetry call
+	// Byte mask to look only at the command field to see if it is 0x10 (telem call)
+	if ((command && 0x0000FF0000) == 0x0000100000) {
+		comm_length = BATTERY_TELEM_COMMAND_LENGTH;
+		comm_delay = BATTERY_I2C_TELEM_DELAY;
+	}
+	else {
+		comm_length = BATTERY_COMMAND_LENGTH;
+		comm_delay = BATTERY_I2C_DELAY;
+	}
+
+	// tell Battery to give us data; and store into our internal buffer
+	int error = i2cTalk(BATTERY_I2C_SLAVE_ADDR, comm_length, BATTERY_RESPONSE_LENGTH,
+							command, response, comm_delay);
+
+	// return error? if an error occurs, else send the data back
+	// TODO: Make this return 0 on a failure? Printf the error instead? Get advice for this
+	if (error != 0) {
+		return error;
+	}
+	else {
```

Comment:
If an `if` condition returns, it does't need an else. Change to:

```c
if (...)
    return error;

return 0;
```

And of course see earlier comments regarding merging with alpha and using `SUCCESS` instead of `0`